### PR TITLE
Transfer Learning for Deep Learning

### PR DESCRIPTION
**Pull request by collins-a**
Deep transfer learning article as index.md
Hero image as hero.jpg

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tool:** [Grammarly](https://grammarly.com)
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by collins-a** [comment]
This fixes #596

File: articles/transfer-learning-with-deep-learning/index.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+### Prerequisites
+
+Basic understanding of machine learning and deep learning. For an introduction or refresher on some basic machine learning concepts check out this [article](/engineering-education/supervised-learning-algorithms/). Some machine learning and deep learning concepts are also covered [here](/engineering-education/automated-fake-news-detection/). For an introduction to transfer learning, I would advise going through my previous [article](https://www.section.io/engineering-education/basics-of-transfer-learning/).
+
+### Useful Terms
+
+**Pre-trained models** – models trained on a sizeable benchmark dataset to solve a problem similar to an already solved problem.
+
+**Marginal probability** – The probability of an event in the presence of all outcomes of another random variable.
```
Comment (lalith1403) [inline articles/transfer-learning-with-deep-learning/index.md:25]: This needs some more clarity. Let's break it down. Marginal probability is the total probability of an RV given conditions on the other random variables. Correct? So what you could do is explain it from the context it will be used in this article.

File: articles/transfer-learning-with-deep-learning/index.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+**Feature** – an individual measurable property that is being observed. It is an attribute shared by all the independent units on that analysis is to be done.
+
+**Loss function** – component of a neural network responsible for calculating loss. Loss refers to the prediction error of a neural network.
```
Comment (lalith1403) [inline articles/transfer-learning-with-deep-learning/index.md:29]: Don't unsupervised learning algorithms also have loss functions? In such cases, they are called the objective function or cost function. What are we trying to optimize? Therefore, it will be wrong to define the loss wrt the prediction error. The prediction error is in the case of supervised learning. Can you re-check the definition?

File: articles/transfer-learning-with-deep-learning/index.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+#### Transfer Learning vs Traditional Machine Learning
+
+To understand transfer learning, it is important to compare it to a traditional machine learning approach. A key difference between the two is isolation. Traditional machine learning can be described as isolated. First things first, it focuses on specific tasks. It also occurs on specific datasets. Another characteristic of the traditional approach is the training of isolated models for very specific tasks. All this means that there is no knowledge that is either retained or accumulated. The learning process takes place without any relationship with past knowledge in different tasks.
```
Comment (lalith1403) [inline articles/transfer-learning-with-deep-learning/index.md:41]: We need to rephrase: It also occurs on specific datasets. What does occur mean in this case?

File: articles/transfer-learning-with-deep-learning/index.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+However, transfer learning takes the contrary approach to the isolated nature of traditional machine learning approaches. The basis of this approach is the transferability of prior knowledge to tackle new tasks. There is no isolation. It is possible to leverage knowledge from pre-trained models to train new models. Here, learning of a new task is dependent on tasks that were learned previously. The result of such is a faster and more efficient learning process.
```
Comment (lalith1403) [inline articles/transfer-learning-with-deep-learning/index.md:43]: What do you mean by-> There is no isolation? Some more emphasis on that would be helpful.

File: articles/transfer-learning-with-deep-learning/index.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+Taking the above into account, transfer learning reduces the training stage of models which results in the use of computational resources more efficiently.
+
+### Types of Deep Transfer Learning
+
+It’s worth noting that, depending on the literature you may come across, the concepts detailed below are related to, or heavily involved in the concept of transfer learning. Some can also be used independently from transfer learning.
+
+#### Domain Adaptation
+
+Domain adaptation involves a scenario where marginal probabilities of the source domain and target domain differ.
```
Comment (lalith1403) [inline articles/transfer-learning-with-deep-learning/index.md:69]: Yes, so the marginal probabilities P(X) = Summation X in P(X,Y). You can express the definition above as follows: The X is the same, whereas the Y's are varying. I have taken a 2-D joint PDF as an example. You may choose whatever you feel apt.

File: articles/transfer-learning-with-deep-learning/index.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+But why? The goal is to add an objective to the model at the source to encourage similarity with the target by confusing the source domain itself. Specifically, domain confusion loss is used to confuse the high-level classification layers of a neural network by matching the distributions of the target and source domains.
+
+A fitting example of such a network is a Domain-Adversarial Neural Network (DANN). The network and relevant concepts around it are expressed in the paper [Domain-Adversarial Training of Neural Networks](https://arxiv.org/abs/1505.07818). DANNs consist of two losses; classification loss and domain confusion loss. The aim is to make sure samples come across as mutually indistinguishable to the classifier. To achieve this, one has to minimize the classification loss for the source samples. One has to also minimize the domain confusion loss for all samples.
+
+#### Multitask Learning
```
Comment (lalith1403) [inline articles/transfer-learning-with-deep-learning/index.md:85]: Can you give an example here? Including a link to your article on GPT-3 would make sense.

File: articles/transfer-learning-with-deep-learning/index.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+In such scenarios, one-shot learning may be of great benefit. One-shot learning is a form of transfer learning where the required output of an algorithm is inferred based on one training example. A model is fed with a single example of the task that is required. For example, a language translation task. A model may be required to translate a word from Swahili to English. It is given a single example of the translation task. Then it is expected to learn from that example for future tasks. This is similar to few-shot learning which is a technique that uses a few examples to generate the required output.
+
+#### Zero-Shot Learning
+
+Zero-shot learning can be useful when dealing with the two scenarios we mentioned when describing one-shot learning. This variant of transfer learning is not dependent on any labeled examples to learn a task. This is training algorithms to carry out tasks they weren’t explicitly trained to do.
```
Comment (lalith1403) [inline articles/transfer-learning-with-deep-learning/index.md:101]: what are the two scenarios?

File: articles/transfer-learning-with-deep-learning/index.md (hunk 6 above)
Comment (lalith1403) [reply articles/transfer-learning-with-deep-learning/index.md:85]: Also, only after reading your article did I realize GPT-3 is a zero-shot learning algorithm. Can you differentiate between the two? How is multi-task learning different from GPT-3? The question arises because GPT-3 is a multi-task model.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
@collins-a Amazing article. Learned a lot. Just a few suggestions that would enhance the readability of the article. As mentioned earlier, I ask questions during the review process. A reader would be going through such questions and we want to make sure we give the reader such answers as well. 
Once again, very well written article. Excited to read the updated article. Cheers 💯

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by collins-a** [comment]
Hey, @lalith1403! Thanks for the ever-refreshing insight. I've made a few requested tweaks. Let me know what you think. Cheers

**Comment by hectorkambow** [comment]
@collins-a 
Thank you for submitting another great article - I keep learning more about AI and ML - 
If you could please go over my edits for this article -as I deleted quite a bit that I felt was repetitive.

Let's be sure we are bringing new applicable knowledge every time to the community. 

This article although I learned about transfer learning - I felt as if it was repeating quite a lot for your previous article ( but without adding much new info on the current topic at hand).

Just wanted to make sure we were aware of that moving forward 🚀  
Thank you

**Comment by collins-a** [comment]
Hey @hectorkambow, thank you for the heads up. I'll definitely step it up going forward.
